src/parse-html.js

```diff
@@ -18,58 +18,64 @@ try {
   d.appendChild(parsed)
   if (d.firstChild.classList[0] !== testClass) throw new Error('Parsed classes not preserved')
   supportsHTMLType = true
+} catch (e) {}
+
+var mockDoc = document.implementation.createHTMLDocument('')
```

Comment:
We need to take all this code out of the try/catch so the fallback function has the flags it needs